wolfcrypt/src/asn.c

```diff
@@ -3703,11 +3692,22 @@ static int ConfirmSignature(const byte* buf, word32 bufSz,
             else {
                 XMEMCPY(plain, sig, sigSz);
 
-                if ((verifySz = wc_RsaSSL_VerifyInline(plain, sigSz, &out,
-                                                                 pubKey)) < 0) {
+                ret = 0;
+                do {
+                #if defined(WOLFSSL_ASYNC_CRYPT)
+                    ret = wc_RsaAsyncWait(ret, pubKey);
+                #endif
+                    if (ret >= 0) {
+                        ret = wc_RsaSSL_VerifyInline(plain, sigSz, &out,
+                                                                    pubKey);
+                    }
+                } while (ret == WC_PENDING_E);
```

Comment:
Correct because this area of the code cannot handle async WC_PENDING_E, so it blocks to complete here, which retains current behavior. Eventually this will need refactored to support async.